test/e2e_node/memory_manager_test.go

```diff
@@ -305,8 +305,42 @@ var _ = SIGDescribe("Memory Manager [Serial] [Feature:MemoryManager][NodeAlphaFe
 
 		// allocate hugepages
 		if *is2MiHugepagesSupported {
-			err := configureHugePages(hugepagesSize2M, 256)
-			framework.ExpectNoError(err)
+			hugepagesCount := 128
+			ginkgo.By("Configuring hugepages")
+			gomega.Eventually(func() error {
+				if err := configureHugePages(hugepagesSize2M, hugepagesCount); err != nil {
+					return err
+				}
+				return nil
+			}, 30*time.Second, framework.Poll).Should(gomega.BeNil())
+
+			ginkgo.By("restarting kubelet to pick up pre-allocated hugepages")
+			restartKubelet()
+
+			ginkgo.By("Waiting for hugepages resource to become available on the local node")
+			gomega.Eventually(func() error {
+				node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), framework.TestContext.NodeName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				capacity, ok := node.Status.Capacity[v1.ResourceName(hugepagesResourceName2Mi)]
+				if !ok {
+					return fmt.Errorf("the node does not have the resource %s", hugepagesResourceName2Mi)
+				}
+
+				size, succeed := capacity.AsInt64()
+				if !succeed {
+					return fmt.Errorf("failed to convert quantity to int64")
+				}
+
+				// 256 Mb
+				expectedSize := int64(hugepagesCount * hugepagesSize2M * 1024)
```

Comment:
Can we add a comment to make it more clear that the expected size is in bytes?

Reply:
done